src/streaming.ts

```diff
@@ -46,46 +52,116 @@ export class StreamProcessor {
     this.eventBus = eventBus;
     this.repository = repository;
     this.log = this.options.logger;
+
+    const minDelay = 5000;
+    const maxDelay = 10000;
+    this.retryAttempt = 1;
+    this.retryDelayMs = Math.floor(
+      Math.random() * (maxDelay - minDelay + 1) + minDelay,
+    );
   }
 
   start(): void {
-    this.log.info('Starting StreamProcessor');
-    const eventSource = new EventSource(
-      `${this.options.baseUrl}/stream?cluster=${this.cluster}`,
-      {
-        headers: {
-          Authorization: `Bearer ${this.jwtToken}`,
-          'API-Key': this.apiKey,
-        },
+    this.log.info('Starting new StreamProcessor');
+
+    const url = `${this.options.baseUrl}/stream?cluster=${this.cluster}`;
+
+    const options = {
+      headers: {
+        'Cache-Control': 'no-cache',
+        Accept: 'text/event-stream',
+        Authorization: `Bearer ${this.jwtToken}`,
+        'API-Key': this.apiKey,
       },
-    );
+    };
 
-    eventSource.onopen = (event: MessageEvent) => {
-      this.log.debug('Stream connected', event);
+    const connected = () => {
+      this.log.info(`SSE stream connected OK: ${url}`);
+      this.retryAttempt = 1;
+      this.readyState = StreamProcessor.CONNECTED;
       this.eventBus.emit(StreamEvent.CONNECTED);
-      // In case we've had to retry, set it back to false
-      this.isRetrying = false
     };
 
-    eventSource.onretrying = (event: MessageEvent) => {
-      this.log.error('Stream is trying to reconnect', event);
-      // Only inform the client once that we're retrying
-      if (!this.isRetrying) {
-        this.isRetrying = true
-        this.eventBus.emit(StreamEvent.RETRYING);
-      }
-    };
+    const failed = (msg) => {
```

Comment:
If we know the type, we should add it here.
```suggestion
    const failed = (msg: string): void => {
```